src/tests/unit/platform/test_admin_grants.py

```diff
@@ -0,0 +1,209 @@
+        ret = copy.deepcopy(POST_PROFILE_IN_GRANT_RESP)
+        ret['profile_uuid'] = body['profile_uuid']
+        return ret
+
+    cbcsdk_mock.mock_request('POST', '/access/v2/orgs/test/grants', POST_GRANT_RESP)
+    cbcsdk_mock.mock_request('POST', '/access/v2/orgs/test/grants/psc:user:ABC12345:DEF67890/profiles',
+                             respond_to_profile_grant)
+    api = cbcsdk_mock.api
+    grant = Grant.create(api, 'psc:user:ABC12345:DEF67890')
+    grant.roles = ["psc:role::SECOPS_ROLE_MANAGER"]
+    grant.org_ref = 'psc:org:test'
+    grant.principal_name = 'Doug Jones'
+    builder = grant.create_profile().add_org("psc:org:test2")
+    profile = builder.add_role("psc:role::SECOPS_ROLE_MANAGER").build()
+    grant.save()
+    assert grant.roles == ["psc:role::SECOPS_ROLE_MANAGER"]
+    assert profile.orgs['allow'] == ["psc:org:test2"]
+
+
+def test_create_new_grant_alt1(cbcsdk_mock):
+    """Test creation of a grant and the profile within the grant, using alternate strategies for the ProfileBuilder."""
+    def respond_to_profile_grant(url, body, **kwargs):
+        ret = copy.deepcopy(POST_PROFILE_IN_GRANT_RESP)
+        ret['profile_uuid'] = body['profile_uuid']
+        return ret
+
+    cbcsdk_mock.mock_request('POST', '/access/v2/orgs/test/grants', POST_GRANT_RESP)
+    cbcsdk_mock.mock_request('POST', '/access/v2/orgs/test/grants/psc:user:ABC12345:DEF67890/profiles',
+                             respond_to_profile_grant)
+    api = cbcsdk_mock.api
+    grant = Grant.create(api, 'psc:user:ABC12345:DEF67890')
+    grant.roles = ["psc:role::SECOPS_ROLE_MANAGER"]
+    grant.org_ref = 'psc:org:test'
+    grant.principal_name = 'Doug Jones'
+    builder = grant.create_profile().set_orgs(["psc:org:test2"]).set_roles(["psc:role::SECOPS_ROLE_MANAGER"])
+    profile = builder.set_disabled(False).build()
+    grant.save()
+    assert grant.roles == ["psc:role::SECOPS_ROLE_MANAGER"]
+    assert profile.orgs['allow'] == ["psc:org:test2"]
+
+
+def test_create_new_grant_fail(cb):
+    """Test failure of the creation of a new grant."""
+    grant = Grant(cb, None)
+    grant.roles = ["psc:role::SECOPS_ROLE_MANAGER"]
+    grant.org_ref = 'psc:org:test'
+    with pytest.raises(ApiError):
+        grant.save()
+
+
+def test_create_profile_from_template(cbcsdk_mock):
+    """Test the creation of a new profile within a grant via a template."""
+    def respond_to_profile_grant(url, body, **kwargs):
+        ret = copy.deepcopy(POST_PROFILE_IN_GRANT_RESP)
+        ret['profile_uuid'] = body['profile_uuid']
+        return ret
+
+    cbcsdk_mock.mock_request('GET', '/access/v2/orgs/test/grants/psc:user:12345678:ABCDEFGH', GET_GRANT_RESP)
+    cbcsdk_mock.mock_request('POST', '/access/v2/orgs/test/grants/psc:user:ABC12345:DEF67890/profiles',
+                             respond_to_profile_grant)
+    api = cbcsdk_mock.api
+    grant = Grant(api, 'psc:user:12345678:ABCDEFGH')
+    template = {
+        "profile_uuid": "to-be-deleted",
```

Comment:
The profile_uuid should not be specified in the template as that is created by the backend

Reply:
No, it shouldn't be, and, if you look at the code for `create_profile()`, the profile_uuid gets explicitly stripped out if it exists.  We have it in the template for this test so that the stripping-out can be tested.

Reply:
Then please add a comment indicating that this is being tested this way

Reply:
Adding this.